threedi_models_and_simulations/api_calls/threedi_calls.py

```diff
@@ -914,9 +914,16 @@ def fetch_schematisation_revision_3di_models(self, schematisation_pk: int, revis
         threedi_models = self.threedi_api.schematisations_revisions_threedimodels(revision_pk, schematisation_pk)
         return threedi_models
 
-    def create_schematisation_revision_3di_model(self, schematisation_pk: int, revision_pk: int) -> ThreediModel:
+    def create_schematisation_revision_3di_model(self, schematisation_pk: int, revision_pk: int, inherit_templates: bool) -> ThreediModel:
         """Create a new 3Di model out of committed revision."""
-        threedi_model = self.threedi_api.schematisations_revisions_create_threedimodel(revision_pk, schematisation_pk)
+        if inherit_templates:
+            data = {
+                "inherit_from_previous_threedimodel": True,
+                "inherit_from_previous_revision": inherit_templates,
+            }
+            threedi_model = self.threedi_api.schematisations_revisions_create_threedimodel(revision_pk, schematisation_pk, data)
+        else:
+            threedi_model = self.threedi_api.schematisations_revisions_create_threedimodel(revision_pk, schematisation_pk)
```

Comment:
After looking at the latest `threedi_ap_client` master branch it seems that `schematisations_revisions_create_threedimodel` method `data` argument is mandatory. Shouldn't then the `data` argument be passed when `inherit_templates is False` as well?